### PR TITLE
refactor(cdk/overlay): use index files for public Sass API instead of copying files out

### PR DESCRIPTION
**Pull request by crisbeto**
Currently we copy out a few of the CDK .scss files to the root of the `cdk` package so that they're more convenient to consume. The problem with this is that it prevents us from importing other Sass files, because the import path will be wrong after the files are copied (see #22043).

These changes rename the files to `_index.scss` which will allow users to maintain the same import paths while allowing us to remove the file copying logic and not having to worry about import paths.

**Note:** the commit message mentions `overlay` due to the linting rules, but this also applies to `a11y` and `text-field`.

BREAKING CHANGES:
Deep imports to the following CDK .scss files need to be updated:
* `@import '~@angular/cdk/overlay/overlay'` -> `@import ~@angular/cdk/overlay`
* `@import '~@angular/cdk/a11y/a11y'` -> `@import ~@angular/cdk/a11y`
* `@import '~@angular/cdk/text-field/text-field'` -> `@import ~@angular/cdk/text-field`

### CODE REVIEW

File: src/cdk/BUILD.bazel [hunk 1]
```diff
@@ -34,20 +26,20 @@ rerootedStyles = [file for target in CDK_ENTRYPOINTS_WITH_STYLES for file in [
 
 # Create genrules which re-root stylesheets from secondary entry-points.
```
Comment (devversion) [inline src/cdk/BUILD.bazel:27]:
```suggestion
# Create genrules that output prebuilt stylesheets for secondary entry-points with styles.
```

**Review by devversion — APPROVED** [review]
LGTM. this is definitely better! One thing to note is that it will break if people did use something like:

```scss
@use `~@angular/cdk/a11y.scss`
```

Though that is probably really uncommon (but maybe worth noting at least)?

**Review by jelbourn — APPROVED** [review]
LGTM- can you amend your commit with a breaking change message for the case of importing with the `.scss` extension?

**Review by mmalerba — APPROVED** [review]

**Comment by crisbeto** [comment]
Done. Also resolved the merge conflicts.

**Pull request merged**